scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3038,6 +3047,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
             blockStatSeq() match {
               case List(q: Quasi) => q.become[Term.Quasi]
               case List(term: Term) => term
+              case Nil => emptyPos(Term.Block(Nil))
```

Comment:
This should not be an issue aside from case clauses, because in most other indented places we can't get an empty Block ended up by outdentation (which comes here from match).